config/routes.rb

```diff
@@ -3,8 +3,17 @@
 Rails.application.routes.draw do
 
   resources :users, only: %i[index show] do
-    resources :posts, only: %i[index show]
+    resources :posts, only: %i[index create new show]
   end
 
+  resources :posts do
+    resources :comments, only: [:create]
+    resources :likes, only: [:create]
+  end
+
+  get '/posts/new', to: 'posts#new'
+  post '/posts/create', to: 'posts#create'
```

Comment:
- [ ] You already have these routes at line 6. No need to create again. To make clean code please delete it.